core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/rollup/RollupServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+                3600, session.getSession().prepare(getFinderCQL(3600)));
+    }
+
+    private String getInsertCQL(int rollup) {
+        return "INSERT INTO rollup" + rollup + " (tenant_id, metric, shard, time, min, max, avg, median, sum, " +
+                "samples, percentiles) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
+    }
+
+    private String getFinderCQL(int rollup) {
+        return "SELECT time, min, max, avg, median, sum, samples, percentiles FROM rollup" + rollup + " " +
+                "WHERE tenant_id = ? AND metric = ? AND shard = 0 AND time >= ? AND time < ?";
+    }
+
+    @Override
+    public Completable insert(MetricId<Double> id, NumericBucketPoint dataPoint, int rollup) {
+        PreparedStatement insert = inserts.get(rollup);
+        checkNotNull(insert, "There is no " + rollup + " rollup");
+        return session.execute(insert.bind(id.getTenantId(), id.getName(), 0L, new Date(dataPoint.getStart()),
+                dataPoint.getMin(), dataPoint.getMax(), dataPoint.getAvg(), dataPoint.getMedian(), dataPoint.getSum(),
+                dataPoint.getSamples(), toMap(dataPoint.getPercentiles()))).toCompletable();
+    }
+
+    @Override
+    public Observable<NumericBucketPoint> find(MetricId<Double> id, long start, long end, int rollup) {
+        PreparedStatement finder = finders.get(rollup);
+        checkNotNull(finder, "There is no " + rollup + " rollup");
+        return session.execute(finder.bind(id.getTenantId(), id.getName(), new Date(start), new Date(end)))
+                .flatMap(Observable::from)
+                .map(row -> new NumericBucketPoint(
+                        row.getTimestamp(0).getTime(),
+                        row.getTimestamp(0).getTime() + TimeUnit.SECONDS.toMillis(rollup),
+                        row.getDouble(1),
+                        row.getDouble(3),
+                        row.getDouble(4),
+                        row.getDouble(2),
+                        row.getDouble(5),
+                        getPercentiles(row.getMap(7, Float.class, Double.class)),
+                        row.getInt(6)));
+    }
+
+    private List<Percentile> getPercentiles(Map<Float, Double> map) {
+        return map.entrySet().stream().map(entry -> new Percentile(entry.getKey().toString(), entry.getValue()))
+                .collect(Collectors.toList());
+    }
+
+    private Map<Float, Double> toMap(List<Percentile> percentiles) {
+        Map<Float, Double> map = new HashMap<>();
+        percentiles.forEach(p -> map.put((float) p.getQuantile(), p.getValue()));
+        return map;
+    }
+
+    @Override
+    public Observable<List<Rollup>> getRollups(MetricId<Double> id, int rawTTL) {
+        return Observable.just(asList(
+                new Rollup(RollupBucket.ROLLUP60.getDuration(), rawTTL * 2),
+                new Rollup(RollupBucket.ROLLUP300.getDuration(), rawTTL * 4),
+                new Rollup(RollupBucket.ROLLUP3600.getDuration(), rawTTL * 6)
```

Comment:
Consider the `getRollups` method throw away code. I just wanted to have something in place quickly for some prototyping. There needs to be reasonable data retention defaults, and they also need to be configurable.